### PR TITLE
Add alter table tests for null shard key tables

### PR DESCRIPTION
**Pull request by agedemenli**
* alter table add/drop/alter_type column (using sequences/different data types/identity columns)
* alter table add constraint (not null, check, exclusion constraint)
* indexes / constraints without names
* multiple subcommands
* create/test/drop view that depends on a dist table with no shard key
* policy & rls test
* alter table add column with a default value / set default / drop default
* alter table set option (autovacuum)

### CODE REVIEW

File: src/test/regress/expected/alter_table_null_dist_key.out [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+INSERT INTO null_dist_table VALUES (NULL, 3, 'test');
+SELECT * FROM null_dist_table;
+  b   | c |       d
+---------------------------------------------------------------------
+ test | 1 | this is a text
+      | 3 | test
+(2 rows)
+
+-- add exclusion constraint
+ALTER TABLE null_dist_table ADD CONSTRAINT exc_b EXCLUDE USING btree (b with =);
```
Comment (onurctirtir) [inline src/test/regress/expected/alter_table_null_dist_key.out:62]: 1. Can we issue rename constraint command right after this?

2. Let's also please add an unnamed version of it:
```sql
ALTER TABLE null_dist_table ADD EXCLUDE USING btree (b with =);
```

File: src/test/regress/citus_tests/config.py [hunk 2]
```diff
@@ -230,9 +230,6 @@ def __init__(self, arguments):
             "local_dist_join_load",
             "local_dist_join",
             "arbitrary_configs_recurring_outer_join",
-            # group 7
-            "sequences_create",
-            "sequences",
```
Comment (onurctirtir) [inline src/test/regress/citus_tests/config.py:235]: as we've discussed earlier, let's not remove this since it still fails due to limited planner support

File: src/test/regress/multi_1_schedule [hunk 3]
```diff
@@ -33,6 +33,7 @@ test: ref_citus_local_fkeys
 test: alter_database_owner
 test: distributed_triggers
 test: create_null_dist_key
+test: alter_table_null_dist_key
```
Comment (onurctirtir) [inline src/test/regress/multi_1_schedule:36]: Usually we place the tests that require add / remove coordinator step but this test is not one of them. And given that this test is a short one, can we parallelize this with another in this schedule?

File: src/test/regress/expected/alter_table_null_dist_key.out [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+DETAIL:  Failing row contains (null, 2, test).
+CONTEXT:  while executing command on localhost:xxxxx
+-- drop not null constraint and try again
+ALTER TABLE null_dist_table ALTER COLUMN b DROP NOT NULL;
+INSERT INTO null_dist_table VALUES (NULL, 3, 'test');
+SELECT * FROM null_dist_table;
+  b   | c |       d
+---------------------------------------------------------------------
+ test | 1 | this is a text
+      | 3 | test
```
Comment (onurctirtir) [inline src/test/regress/expected/alter_table_null_dist_key.out:58]: let's use an `order by c` or such

File: src/test/regress/expected/alter_table_null_dist_key.out [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+ALTER TABLE null_dist_table ADD CONSTRAINT exc_b EXCLUDE USING btree (b with =);
+-- test multiple subcommands
+ALTER TABLE null_dist_table ADD COLUMN int_column1 INTEGER,
+	DROP COLUMN d;
+SELECT * FROM null_dist_table;
+  b   | c | int_column1
+---------------------------------------------------------------------
+ test | 1 |
+      | 3 |
+(2 rows)
```
Comment (onurctirtir) [inline src/test/regress/expected/alter_table_null_dist_key.out:71]: let's use an `order by c` or such

**Review by onurctirtir — COMMENTED** [review]
Those are my final comments for DDL support. I manually tested some of the cases that we don't have a test for --hence not listed them here-- but listed the critical ones that I think we should definitely have a test for.

1. alter table set options

   A good candidate for columnar, let's set stripe size or something.

2. Policy creation "after" creating a null-shard-key table:
   ```sql
   CREATE TABLE null_dist_key (table_user text);
   SELECT create_distributed_table('null_dist_key', null);

   ALTER TABLE null_dist_key ENABLE ROW LEVEL SECURITY;

   CREATE ROLE table_users;
   CREATE POLICY table_policy ON null_dist_key TO table_users
      USING (table_user = current_user);
  
   // connect back as a different user and try querying the rows inserted by 
   // this user, and see that they fail
   ```

3. create / drop publication / subscription (based on null-shard-key tables)

   This is one of the latest features introduced in the latest release, 
   though I'm not sure how much we should test now / how much should we 
   defer the testing around this to CDC-testing task (1.10).

   I'd suggest to you sync with Marco to hear what he thinks about that.
   (cc: @marcocitus)

**Review by onurctirtir — APPROVED** [review]
Thanks for the good test coverage. Let's not forget to include last couple of tests in the PR description (and hence in the commit message).

We decided to add tests for create/drop publication/subscription commands when testing CDC feature, hence skipping them for now.
